### DISCUSSION TITLE
[3.x] Add setter for custom settings view path

### DISCUSSION THREAD

**Discussion by renepardon** [area17/twill#2460] [Q&A]
## Summary

Right now, the views for settings can only be created inside the laravel app resource path at `resources/views/twill/settings` as defined in `AppSetting#getFormBlocks()`. It should be possible to add additional paths were settings views can be loaded from. Especially if creating packages which registers additional settings specific to that package.

## Describe the solution you'd like

Instead of only reading the views from  resource path with: `$directory = resource_path('views/twill/settings/' . $this->getSettingGroup()->getName());` it should be possible to add additional scan paths dynamically. For example through a static setter or a configuration array which can be extended by a package to add it's own settings view path.

## Additional context

If you tell me what way you prefer to add this functionality, I could provide a pull request for this.

**Comment by raymondtri** [comment]
Have you tried (in the service provider where you're defining settings)

```
use A17\Twill\Facades\TwillAppSettings;

...
public function boot(): void
{
		TwillAppSettings::registerSettingsPath(
			'my/new/custom/view/settings/path'
		);
}
```
